var/spack/repos/builtin/packages/opencv/package.py

```diff
@@ -891,30 +891,38 @@ def determine_version(cls, lib):
             pattern = None
             if ext == "dylib":
                 # Darwin switches the order of the version compared to Linux
-                pattern = re.compile(r"lib(\S*?)_(\S*)\.(\d+\.\d+\.\d+)\.%s" % ext)
+                pattern = re.compile(r"libopencv_(\S*?)\.(\d+\.\d+\.\d+)\.%s" % ext)
             else:
-                pattern = re.compile(r"lib(\S*?)_(\S*)\.%s\.(\d+\.\d+\.\d+)" % ext)
+                pattern = re.compile(r"libopencv_(\S*?)\.%s\.(\d+\.\d+\.\d+)" % ext)
             match = pattern.search(lib)
             if match:
-                ver = match.group(3)
+                ver = match.group(2)
         return ver
 
     @classmethod
     def determine_variants(cls, libs, version_str):
         variants = []
-        remaining_modules = set(Opencv.modules)
+        remaining_modules = set(Opencv.modules + Opencv.contrib_modules)
```

Comment:
Does this work if we use cls instead of Opencv? Feels weird to call out the class by name.

Reply:
good catch. I finally got around to updating the PR